src/connector/src/parser/postgres.rs

```diff
@@ -386,8 +389,8 @@ pub fn postgres_row_to_owned_row(row: tokio_postgres::Row, schema: &Schema) -> O
                                 );
                             }
                         };
+                        Some(ScalarImpl::from(ListValue::new(builder.finish())))
```

Comment:
I tried to understand why this statement is moved from outside to both branches of `if-else` separating array of enums from array of others, and realized it makes error of getting `Vec<EnumString>` return `None` rather than an empty array. Sounds like it applies to all other array types, but can be a breaking change.

Do we want to change the error case from empty array to null for all array types?

Reply:
Currently, **only** enum list with `NULL` will be converted to `NULL` in debezium stream. It's a special case (More like a bug in debezium). Other type lists are converted into string lists in debezium stream and then be parsed item by item in RW, so that the **valid items in a list** with an wrong item **are kept after conversion**. So I think we'd better keep backfilling's behavior aligned with debezium stream? i.e., only enum list with `NULL` will be converted to `NULL` while if a list of other type contains a invalid item, we just set that invalid item as `NULL`.

Reply:
I was not talking about debezium / null / enum. It was about the general case when any call to `row.try_get(i)` returns `Err` for whatever unanticipated reason - The error handling logic fallback to `null` for this column if its data type is non-list or os enum-list, but fallback to `{}` for other lists. This feels inconsistent.

Of course before this PR all non-list fallback to `null` and all lists (incl. enum list, for whatever reason incl. debezium bug of handling nulls inside enum list) fallback to `{}`.

The desired behavior might be to always fallback to `null` regardless of data type.

Reply:
I roughly understand, but there's another issue: from outside this function, how can we distinguish whether the NULL received is fallbacked from an invalid datum or is inherently NULL? Should we directly return an error when encountering an invalid datum?